File: FTAND03U2C1/app/src/main/java/com/example/ft_and_03u_2c_1/SignInActivity.java
package com.example.ft_and_03u_2c_1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class SignInActivity extends AppCompatActivity {

    private Button etBack,etSignInPage,btnSignIn;
    private EditText etNames,etPassword;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sign_in);
        itnitMain();
        etBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(SignInActivity.this, MainActivity.class);
                startActivity(intent);
            }
        });

        etSignInPage.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(SignInActivity.this,SignUpActivity.class);
                startActivity(intent);
            }
        });

        btnSignIn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(isName() && isPassword()){
                    Intent intent = new Intent(SignInActivity.this,HomeActivity.class);
                    intent.putExtra("name",etNames.getText().toString());
                    startActivity(intent);
                }
            }
        });

    }

    private boolean isPassword() {
        boolean b=false;
        if(etPassword.getText().toString().length()>=6 && etPassword.getText().toString().matches("[0-9]+")){
            b=true;
        }else if(etPassword.getText().toString().length()<6){
            etPassword.setError("password is weak");
        }else{
            etPassword.setError("no alphabets allowed");
        }
        return b;
    }

    private boolean isName() {
        boolean b=false;
        String name = etNames.getText().toString();
        if(name.trim().length()>=4){
            b=true;
        }else{
            etNames.setError("minimum 4 characters");
        }
        return  b;
    }

    private void itnitMain() {
        etBack=findViewById(R.id.back);
        etSignInPage=findViewById(R.id.btnSignUpPage);
        etNames=findViewById(R.id.editNames);
        etPassword=findViewById(R.id.editPassword);

        btnSignIn=findViewById(R.id.btnSignIn);
    }
}

File: RoomDatabaseMoneyManagerapp/app/src/main/java/com/example/roomdatabasemoneymanagerapp/SignInActivity.kt
package com.example.roomdatabasemoneymanagerapp

import android.content.Intent
import android.content.res.ColorStateList
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.core.widget.doAfterTextChanged
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.example.roomdatabasemoneymanagerapp.MVVM.Repository.MoneyRepo
import com.example.roomdatabasemoneymanagerapp.MVVM.ViewModell.TaskViewModel
import com.example.roomdatabasemoneymanagerapp.MVVM.ViewModell.TaskViewModelFactory
import com.example.roomdatabasemoneymanagerapp.Retrofit.Status
import com.example.roomdatabasemoneymanagerapp.Retrofit.requests.LoginRequestModel
import com.example.roomdatabasemoneymanagerapp.RoomDataBase.RoomDataBase
import com.example.roomdatabasemoneymanagerapp.RoomDataBase.TaskExpenseDao
import com.example.roomdatabasemoneymanagerapp.RoomDataBase.TaskIncomeDao
import kotlinx.android.synthetic.main.activity_sign_in.*
import org.jetbrains.anko.longToast

class SignInActivity : AppCompatActivity() {

    lateinit var roomDB:RoomDataBase
    lateinit var taskIncomeDao: TaskIncomeDao
    lateinit var taskExpenseDao: TaskExpenseDao
    lateinit var taskViewModel: TaskViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sign_in)

        emailSignIn.doAfterTextChanged {
            if(passwordSignIn.text.toString().length>=6 && emailSignIn.text.toString().length>0) {
//                signInBtn.supportBackgroundTintList
                signInBtn.backgroundTintList = ColorStateList.valueOf(resources.getColor(R.color.black))
            } else {
                signInBtn.backgroundTintList = ColorStateList.valueOf(resources.getColor(R.color.grey))
            }
        }
        passwordSignIn.doAfterTextChanged {
            if(passwordSignIn.text.toString().length>=6 && emailSignIn.text.toString().length>0) {
//                signInBtn.supportBackgroundTintList
                signInBtn.backgroundTintList = ColorStateList.valueOf(resources.getColor(R.color.black))
            } else {
                signInBtn.backgroundTintList = ColorStateList.valueOf(resources.getColor(R.color.grey))
            }
        }

//        emailSignIn.doAfterTextChanged {
//            text.text = emailSignIn.text.toString()
//        }


        roomDB = RoomDataBase.getDatabaseObject(this)
        taskIncomeDao = roomDB.getIncomeDao()
        taskExpenseDao = roomDB.getExpenseDao()
        val repo = MoneyRepo(taskIncomeDao,taskExpenseDao)

        val taskViewModelFactory = TaskViewModelFactory(repo)
        taskViewModel = ViewModelProviders.of(this,taskViewModelFactory).get(TaskViewModel::class.java)


//        val loginRequestModel = LoginRequestModel( userName =
////            "pradeep1706108@gmail.com",password = "dhankhar")


        signInBtn.setOnClickListener {
           val loginRequestModel = LoginRequestModel(emailSignIn.text.toString(),passwordSignIn.text.toString())
//            val loginRequestModel = LoginRequestModel( userName =
//            "pradeep1706108@gmail.com",password = "dhankhar")
            taskViewModel.userLogin(loginRequestModel)
                .observe(this, Observer {

                    val response = it

                    when(response.status){
                        Status.SUCCESS->{
                            val name = response.data?.user?.name!!
                            val email = response.data?.user?.email!!
                            longToast("$name and $email")
                            val intent = Intent(this,MainActivity::class.java)
                            startActivity(intent)
                        }
                        Status.ERROR->{
                            val error = response.message!!
                            longToast("Invalid email and password")
                        }
                        Status.LOADING->{

                        }
                    }
                })
        }

        registerInBtn.setOnClickListener {
            val intent = Intent(this,SignUpActivity::class.java)
            startActivity(intent)
        }

    }
}